text/0000-forwardpropstoheader.md

```diff
@@ -0,0 +1,186 @@
+# Reactive Header Control with `<this.props.navigation.ForwardPropsToHeader>`
+## Problem
+In the React lifecycle, all rendering should be done inside `render()`. But if a screen needs to control how a header is rendered, it cannot do so in the `render()` method. Instead, the Screen must use a third-party object to pass callbacks through. This does not scale well for complex headers.
+
+It is possible to use 
+```jsx
+static navigationOptions = {
+  header: (props: *) => {
+    globalStateObject.setHeaderProps(props)
+    return null
+  },
+}
+render() {
+  return (
+    <View>
+      <CloseUpGallery photos={photos}/>
+      <ViewAlbumHeader
+        onChangeText={console.warn}
+        {...db.headerProps}
+      />
+      {/* Here is the body of the Screen */}
+      <FooComponent/>
+    </View>
+  )
+}
+```
+However, this leaves a lot to be desired. When using `headerMode: 'float'`, two headers will be displayed during the screen transition. The alternative on iOS is having the non-native look of `headerMode: 'screen'`.
+
+## Desire
+```jsx
+class Screen extends React.Component {
+  // ...
+  render() {
+    return (
+      <View>
+        <CustomHeader
```

Comment:
I agree that there are many benefits to having the header rendered inside of the normal render function of the screen component!